src/checkstatus/SSMAutomationExecutionStatusWaiter.py

```diff
@@ -0,0 +1,31 @@
+import botocore
+import constants
+import waiter_acceptor_config
+
+def check_automation_execution_status(automation_execution_id, max_attempts, ssm):
```

Comment:
the defined abstraction seems to just check the execution status , should we just retrun the status then
```
execution = ssm.describe_automation_executions(Filters=automation_execution_filter)['AutomationExecutionMetadataList'][0]
    return execution['AutomationExecutionStatus']
```

Reply:
its handling the waiter logic so the check is done multiple times in the 15 min duration of the lambda execution

Reply:
does this increase the number of API calls internally to AWS and results in cost increase of the lambda? If yes, probably for now we can put a comment stating to configure the delay accordingly

Reply:
it uses the same delay that has been configured for the rest of the state machine
delay: https://github.com/intuit/Trapheus/blob/master/src/common/python/constants.py#L3

max attempts: https://github.com/intuit/Trapheus/blob/master/src/common/python/utility.py#L74

Reply:
yes , add a comment to configure this - i am guessing this would have a cost impact till we move to events

Reply:
one way to optimise this would be to not introduce a waiter on ```DescribeAutomationExecutions``` but just get the list of  automation execution and then use a while loop to until you get the status completed, catch exeception on TLE and retry execution

Reply:
what is the advantage in doing that - considering the waiter is accomplishing the same thing? we dont save on cost on running a while loop

Reply:
no i meant ```DescribeAutomationExecutions``` in SSM has an associated cost that we could save. As per this https://aws.amazon.com/systems-manager/pricing/ its 
```
Get, Describe, Update, and GetOpsSummary API requests | $0.039 per 1,000 requests
```

Reply:
still not understanding how we are saving on cost with the while loop. repeated calls will need to be made for getting the status

Reply:
We can take this as a followup if needed

Reply:
i will check this as a follow up. this option will also need repeated invocations till the automation is complete

Reply:
yes but makes the API call just once , i am good with a followup on this